From e9ef77c7b810ff9b7054d1f6dd9532244b421b7f Mon Sep 17 00:00:00 2001
From: Troy Mitchell <troy.mitchell@linux.spacemit.com>
Date: Wed, 29 Apr 2026 09:38:50 +0800
Subject: [PATCH 253/283] FROMLIST: ASoC: dt-bindings: add fixed-sample-rate
 property for SpacemiT K1/K3

Add the optional spacemit,fixed-sample-rate property. When multiple I2S
controllers share a common bclk, this property constrains all
controllers to the same sample rate. This applies to both K1 and K3
SoCs and is only needed when two or more I2S controllers are active
simultaneously.

Signed-off-by: Troy Mitchell <troy.mitchell@linux.spacemit.com>
Link: https://lore.kernel.org/r/20260429-k3-i2s-v1-5-2fe99db11ecb@linux.spacemit.com
Signed-off-by: Han Gao <gaohan@iscas.ac.cn>
---
 .../devicetree/bindings/sound/spacemit,k1-i2s.yaml        | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/Documentation/devicetree/bindings/sound/spacemit,k1-i2s.yaml b/Documentation/devicetree/bindings/sound/spacemit,k1-i2s.yaml
index 240d90402e4f..137070ad9eba 100644
--- a/Documentation/devicetree/bindings/sound/spacemit,k1-i2s.yaml
+++ b/Documentation/devicetree/bindings/sound/spacemit,k1-i2s.yaml
@@ -84,6 +84,14 @@ properties:
   "#sound-dai-cells":
     const: 0
 
+  spacemit,fixed-sample-rate:
+    $ref: /schemas/types.yaml#/definitions/uint32
+    description:
+      Fixed sample rate in Hz. When multiple I2S controllers share a
+      common bclk, this property constrains all controllers to the same
+      sample rate. Only needed when two or more I2S controllers are
+      active simultaneously.
+
 required:
   - compatible
   - reg
-- 
2.47.3

